system/CLI/CLI.php

```diff
@@ -750,6 +750,121 @@ public static function getOptionString(): string
 	}
 
 	//--------------------------------------------------------------------
+
+	/**
+	 * Returns a well formated table
+	 *
+	 * @param  array  $tbody List of rows
+	 *         			[
+	 *         				['id' => 1, 'foo' => 'bar']
+	 *         			]
+	 * @param  array  $thead List of columns
+	 *         			['ID', 'Foo']
+	 *
+	 * @return string
+	 *
+	 * +----+-----+
+	 * | ID | Foo |
+	 * +----+-----+
+	 * | 1  | bar |
+	 * +----+-----+
+	 */
+	public static function table(array $tbody, array $thead = [])
+	{
+		// All the rows in the table will be here until the end
+		$table_rows = [];
+
+		// We need only indexes and not keys
+		if (! empty($thead))
+		{
+			$table_rows[] = array_values($thead);
+		}
+
+		foreach ($tbody as $tr)
+		{
+			$table_rows[] = array_values($tr);
+		}
+
+		// Yes, it really is necessary to know this count
+		$total_rows = count($table_rows);
+
+		// Store all columns lengths
+		// $all_cols_lengths[row][column] = length
+		$all_cols_lengths = [];
+
+		// Store maximum lengths by column
+		// $max_cols_lengths[column] = length
+		$max_cols_lengths = [];
+
+		// Read row by row and define the longest columns
+		for ($i = 0; $i < $total_rows; $i++)
+		{
+			$column = 0; // Current column index
+			foreach ($table_rows[$i] as $col)
+			{
+				// Sets the size of this column in the current row
+				$all_cols_lengths[$i][$column] = strlen($col);
+
+				// If the current column does not have a value among the larger ones
+				// or the value of this is greater than the existing one
+				// then, now, this assumes the maximum length
+				if (! isset($max_cols_lengths[$column]) || $all_cols_lengths[$i][$column] > $max_cols_lengths[$column])
+				{
+					$max_cols_lengths[$column] = $all_cols_lengths[$i][$column];
+				}
+
+				// We can go check the size of the next column...
+				$column++;
+			}
+		}
+
+		// Read row by row and add spaces at the end of the columns
+		// to match the exact column length
+		for ($i = 0; $i < $total_rows; $i++)
+		{
+			$column = 0;
+			foreach ($table_rows[$i] as $col)
+			{
+				$diff = $max_cols_lengths[$column] - strlen($col);
+				if ($diff)
+				{
+					$table_rows[$i][$column] = $table_rows[$i][$column] . str_repeat(' ', $diff);
+				}
+				$column++;
+			}
+		}
+
+		$table = '';
+
+		// Joins columns and append the well formatted rows to the table
+		for ($i = 0; $i < $total_rows; $i++)
+		{
+			// Set the table border-top
+			if ($i === 0)
+			{
+				$cols = '+';
+				foreach ($table_rows[$i] as $col)
+				{
+					$cols .= str_repeat('-', strlen($col) + 2) . '+';
+				}
+				$table .= $cols . PHP_EOL;
+			}
+
+			// Set the columns borders
+			$table .= '| ' . implode(' | ', $table_rows[$i]) . ' |' . PHP_EOL;
+
+			// Set the thead and table borders-bottom
+			if ($i === 0 && ! empty($thead) || $i + 1 === $total_rows)
+			{
+				$table .= $cols . PHP_EOL;
+			}
+		}
+
+		//fwrite(STDOUT, $table . PHP_EOL);
+		return $table;
```

Comment:
The intention here is to be able to show the output and not return but when using `fwrite()` I am not being able to make the test pass ...

I tried with `ob_start()` and nothing... Even no success with `CLITestStreamFilter::$buffer`...

Reply:
Saned.